homeassistant/components/smhi/config_flow.py

```diff
@@ -0,0 +1,145 @@
+def smhi_locations(hass: HomeAssistant):
+    """Return configurations of smhi component."""
+    return set((slugify(entry.data[CONF_NAME])) for
+               entry in hass.config_entries.async_entries(DOMAIN))
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class SmhiFlowHandler(data_entry_flow.FlowHandler):
+    """Smhi config flow."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize smhi forecast configuration flow."""
+        self._errors = {}
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        return await self.async_step_init(user_input)
+
+    async def async_step_init(self, user_input=None):
+        """Handle the flow logic for SMHI location config."""
+        self._errors = {}
+
+        # If hass config has the location set and
+        # it is not configured already, the default
+        # location is created at home coordinates
+        if self._homeassistant_location_exists() and \
+           not self._home_location_exists():
+            return await self._create_home_location_entry()
+
+        # Add the location if not already exists
+        if user_input is not None:
+            is_ok = await self._check_location(
+                user_input[CONF_LONGITUDE],
+                user_input[CONF_LATITUDE]
+            )
+            if is_ok:
+                name = slugify(user_input[CONF_NAME])
+                if not self._name_in_configuration_exists(name):
+                    return self.async_create_entry(
+                        title=user_input[CONF_NAME],
+                        data=user_input,
+                    )
+
+                self._errors['base'] = 'name_exists'
+            else:
+                self._errors['base'] = 'wrong_location'
+
+        return await self._show_config_form()
+
+    def _homeassistant_location_exists(self) -> bool:
+        """Return true if default location is set."""
+        if self.hass.config.latitude != 0.0 and \
+           self.hass.config.longitude != 0.0:
+            return True
+        return False
+
+    def _home_location_exists(self) -> bool:
+        """Return True if home location exists."""
+        if HOME_LOCATION_NAME in smhi_locations(self.hass):
+            return True
+        return False
+
+    def _name_in_configuration_exists(self, name: str) -> bool:
+        """Return True if name exists in configuration."""
+        if name in smhi_locations(self.hass):
+            return True
+        return False
+
+    async def _create_home_location_entry(self) -> Dict:
+        """Create default home location entry."""
+        return self.async_create_entry(
+            title=self.hass.config.location_name,
+            data={
+                CONF_NAME: self.hass.config.location_name,
+                CONF_LATITUDE: self.hass.config.latitude,
+                CONF_LONGITUDE: self.hass.config.longitude
+            })
+
+    async def _show_config_form(self):
```

Comment:
This helper doesn't seem necessary.

Reply:
It is to make testing easier, one less thing to mock away and easier to unit test. Think I keep this one